src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -387,5 +391,52 @@ get_kernel_freq(const axlf* top)
   return kernel_clk_freq;
 }
 
+
+std::vector<uint64_t>
+get_kernel_inst_addrs(const axlf* top)
+{
+  std::vector<uint64_t> addrVec;
+  const axlf_section_header *xml_hdr = ::xclbin::get_axlf_section(top, EMBEDDED_METADATA);
+  if (xml_hdr) {
+    auto begin = reinterpret_cast<const char*>(top) + xml_hdr->m_sectionOffset;
+    const char *xml_data = reinterpret_cast<const char*>(begin);
+    uint64_t xml_size = xml_hdr->m_sectionSize;
+
+    pt::ptree xml_project;
+    std::stringstream xml_stream;
+    xml_stream.write(xml_data, xml_size);
+    pt::read_xml(xml_stream, xml_project);
+
+    auto xml_plotform = xml_project.get_child_optional("project.platform");
+
+    if (xml_plotform) {
+      for (auto& xml_device : xml_project.get_child("project.platform")) {
+        if (xml_device.first != "device")
+          continue;
+        for (auto& xml_core : xml_device.second) {
+          if (xml_core.first != "core")
+            continue;
+          for (auto& xml_kernel : xml_core.second) {
+            if (xml_kernel.first != "kernel")
+              continue;
+            for (auto& xml_inst : xml_kernel.second) {
+              if (xml_inst.first != "instance")
+                continue;
+              auto name = xml_inst.second.get<std::string>("<xmlattr>.name");
+              for (auto& xml_remap : xml_inst.second) {
+                if (xml_remap.first != "addrRemap")
+                  continue;
+                auto base = convert(xml_remap.second.get<std::string>("<xmlattr>.base"));
+                addrVec.push_back(base);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
```

Comment:
Too many levels of nesting. Please restructure with early exits.
For example, first two levels of indent can be eliminated with
```
if (!xml_hdr)
  return addrVec;
...
if (!xml_platform)
 return addrVec;
```
Any indents after that can be left as is, though factoring out into separate embedded lambda functions may be cleaner; don't worry about this though.

Reply:
addressed this comment in the code.